src/pyOpenMS/pyopenms/_dataframes.py

```diff
@@ -516,4 +519,34 @@ def extract(pep):
                 ret.append(default_missing_values[type(val)])
         return tuple(ret)
 
-    return _pd.DataFrame(_np.fromiter((extract(pep) for pep in peps), dtype=dt, count=count))
+    return _pd.DataFrame(_np.fromiter((extract(pep, pep_idx) for pep_idx, pep in enumerate(peps)), dtype=dt, count=count))
+
+
+def update_scores_from_df(peps: List[_PeptideIdentification], df : _pd.DataFrame, main_score_name : str):
+    """Updates the scores in PeptideIdentification objects using a pandas dataframe.
+    Parameters:
```

Comment:
Did you check if this renders properly? We usually go for :param xyz: these days.

Do you need/want to specify the types twice?

Reply:
no clue I just copied the style from the ones above.  I can look into it

Reply:
I think the aliased classes are not documented correctly anyway: https://github.com/OpenMS/pyopenms-docs/issues/326

Reply:
can look into it.